src/design/econsecurity.md

```diff
@@ -1,13 +1,82 @@
 
-<!-- ## todo
 
-* existing transaction fee
 
-* more fee structure examples -->
\ No newline at end of file
+Substrate provides several ways to affect the fees charges for executing a transaction. Substrate developer hub contains full details about [fees](https://substrate.dev/docs/en/next/development/module/fees) and [weights](https://substrate.dev/docs/en/next/conceptual/runtime/weight).
+
+* Base fee - Applies a fixed fee to each and every transaction. A parameter in the `transaction_payment` module.
+
+* Length fee - Applies a fee proportional to the transaction's length in bytes. The constant is a parameter in the `transaction_payment` module.
+
+* Transaction weight - Each transaction can declare a weight, either fixed, or calculated from its parameters. This is exemplified briefly below and more thoroughly in the kitchen.
+
+* Weight to Fee - A function to convert weight to fee. It doesn't need to be linear, although it often is. The same conversion function is applied across all transactions from all modules in the runtime. This is exemplified briefly below and more thoroughly in the kitchen.
+
+## Assigning Transaction Weights
+
+For simple transactions a fixed weight will do.
+```rust
+decl_module! {
+	pub struct Module<T: Trait> for enum Call {
+
+		#[weight = SimpleDispatchInfo::FixedNormal(100)]
+		fn store_value(_origin, entry: u32) -> Result {
+			// --snip--
+		}
+```
+
+For more complex transactions, custom weight calculations can be performed.
+```rust
+pub struct Conditional(u32);
+
+impl WeighData<(&bool, &u32)> for Conditional {
+	fn weigh_data(&self, (switch, val): (&bool, &u32)) -> Weight {
+
+		if *switch {
+			val.saturating_mul(self.0)
+		}
+		else {
+			self.0
+		}
+	}
+}
+```
+
+These examples, and several others can be compiled in the kitchen.
+
+## Converting Weight To Fees
+
+In many cases converting weight to fees 1:1 will suffice and be accomplished with `ConvertInto`. This approach is taken in the [node template](https://github.com/substrate-developer-hub/substrate-node-template/blob/43ee95347b6626580b1d9d554c3c8b77dc85bc01/runtime/src/lib.rs#L230) as well as the kitchen's own super runtime.
+```rust
+impl transaction_payment::Trait for Runtime {
+	// --snip--
+	type WeightToFee = ConvertInto;
+}
+```
+
+This example uses a quadratic conversion and supports custom coefficients
+```rust
+pub struct QuadraticWeightToFee<C0, C1, C2>(C0, C1, C2);
+
+impl<C0, C1, C2> Convert<Weight, Balance> for QuadraticWeightToFee<C0, C1, C2>
+	where C0: Get<Balance>, C1: Get<Balance>, C2: Get<Balance> {
+
+	fn convert(w: Weight) -> Balance {
+		let c0 = C0::get();
+		let c1 = C1::get();
+		let c2 = C2::get();
+		let w = Balance::from(w);
+
+		// TODO use safe math
+		c0 + c1 * w + c2 * w * w
```

Comment:
```suggestion
		c0.saturating_add(c1.saturating_mul(w)).saturating_add(c2.saturating_mul(w).saturating_mul(w))
```

Reply:
Question. I had left this as a TODO in the text portion so that users would see it really is a simple arithmetic calculation. I've actually used the saturating functions in the ktichen portion of the code. Do you think it's better to show the full saturating syntax here?

Reply:
fair enough. can be ignored then.